tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -159,14 +165,22 @@ public static void Initialize()
 
             InitializeNoNativeParts();
             var tracer = Tracer.Instance;
-            InitializeLiveDebugger();
 
             if (tracer is null)
             {
                 Log.Debug("Skipping TraceMethods initialization because Tracer.Instance was null after InitializeNoNativeParts was invoked");
             }
             else
             {
+                try
+                {
+                    InitRemoteConfigurationManagement(tracer);
```

Comment:
I would suggest retrying this asynchronously if it failed initialising, and the same approach for like [354](https://github.com/DataDog/dd-trace-dotnet/pull/3077/files#diff-830094626295c57a4711277c423e320807e5b811816f9f38237b60e75cba088aR354)

Reply:
I agree with you.
I've added simple retry mechanism at 7329b72ccc532070ba31781f0cd2a1a7f74aefee

However, I have a feeling, that `DiscoveryService` should get more effort in future.
For example, now I'm doing only 1 discovery request, and re-using same endpoint discovered.

What if one of endpoint/s will change?
How affected services should react for it?
How dependent services should react, if they are in the middle of some operations?

Those are just simple basic questions, I'm pretty sure there are more will come.
I think it deserve a dedicated PR.

Reply:
Endpoint changes are rare and also not an issue, the agent maintains backward compatibility for a damn longtime.
As there's no way to force an upgrade on tracers in use, the weight is on us maintaining it the APIs.

Reply:
Oh, I see, make sense.
Changed at 4a0357e837f54e71758db977091e7922692b9958

Reply:
LGTM